frontend/portals/CartItemSubscriptionInfo/index.jsx

```diff
@@ -1,40 +1,36 @@
+/* eslint-disable camelcase */
 import React from 'react';
 import PropTypes from 'prop-types';
 import connect from './connector';
 import Label from './components/Label';
-import { getDiscountedPrice } from '../../helpers/rechargeDiscountPriceTools';
-import { NO_SUBSCRIPTION_FREQUENCY_VALUE } from '../../constants';
 
 /**
  * Distill subscriptions array to array of simpler objects
  * @param {Object[]} subscriptions Subscription objects
  * @param {number} itemUnitPrice Item unit price
  * @return {Object[]}
  */
-const distillSubscriptions = (subscriptions, itemUnitPrice) => subscriptions
+const distillSubscriptions = subscriptions => subscriptions
   .filter(subscription => (
     typeof subscription === 'object'
-    && subscription.frequencyValue
-    && subscription.frequencyValue !== NO_SUBSCRIPTION_FREQUENCY_VALUE
-    && subscription.subscriptionInfo
+    && subscription.charge_interval_frequency
+    && subscription.charge_interval_frequency !== null
   ))
   .map((subscription) => {
     const {
-      frequencyValue,
-      subscriptionInfo: {
-        intervalUnit = '',
-        quantity = 0,
-        discountType,
-        discountAmount,
-      },
+      charge_interval_frequency,
+      order_interval_unit,
+      quantity,
+      price,
     } = subscription;
-    const price = getDiscountedPrice(itemUnitPrice, discountType, discountAmount);
     const totalPrice = price * quantity;
 
+    const unitLabel = order_interval_unit === 'day' ? 'days' : order_interval_unit;
```

Comment:
why do you only care about day/days?

Reply:
Added a new helper to address months and weeks as well